src/app/view/components/_common/iframe/iframe-porlet-legacy/iframe-porlet-legacy.component.ts

```diff
@@ -20,17 +21,26 @@ export class IframePortletLegacyComponent implements OnInit {
     url: BehaviorSubject<string> = new BehaviorSubject('');
     isLoading = false;
 
+    private readonly customEventsHandlers;
+
     constructor(
         private contentletService: DotContentletService,
         private dotLoadingIndicatorService: DotLoadingIndicatorService,
         private dotMenuService: DotMenuService,
         private dotRouterService: DotRouterService,
         private dotcmsEventsService: DotcmsEventsService,
-        private ngZone: NgZone,
+        private dotContentletEditorService: DotContentletEditorService,
         private route: ActivatedRoute,
         public loggerService: LoggerService,
         public siteService: SiteService
-    ) {}
+    ) {
+        if (!this.customEventsHandlers) {
+            this.customEventsHandlers = {
+                'edit-page': this.goToEditPage.bind(this),
+                'edit-contentlet': this.editContentlet.bind(this)
+            };
```

Comment:
this can become a lot code, maybe can be better handled this in another class:

```
onCustomEvent($event: CustomEvent): void {
        this.customEventsHandler.handle($event);
}
```